### DISCUSSION TITLE
Allow manual flow button to trigger custom code (for example a hook)

### DISCUSSION THREAD

**Discussion by henk23** [directus/directus#14445] [Archive]
The usecase is something like this:

- User creates an item as draft
- Supervisor can review the item and click on the flow button "approve" (already possible)
- This triggers all kinds of actions (already possible)
  - status is set to published (already works in flows)
  - but also some custom code is executed on the payload. (I believe not possible yet)

This could be as simple as adding a new hook action event "flow.{flow_name}.trigger" that then can be listened to or it could even be an explicit new flow operation: "Trigger Event", allowing to set the name of the hook to be triggered.

**Comment by rijkvanzanten** [comment]
I believe this already exists! You can setup a manual flow trigger, and have it configured to show up on the collection you want to review on. When the user selects items and runs the trigger, the trigger input payload receives the items that where selected when the flow was triggered, allowing you to run all sorts of operations on those items 👍🏻

**Reply by benhaynes** [reply]
Does the sidebar option show up on the Item Detail page too? That would be ideal.

**Reply by rijkvanzanten** [reply]
Yup!

**Reply by henk23** [reply]
Yeeees, all sorts of operations are already possible. But as far as I see there is no way to run my custom code on the payload, like I can do in the extensions, right? I would need to create a custom endpoint and configure a webhook to call that endopoint. My idea was to call my custom code directly.

**Reply by henk23** [reply]
I updated title and description for clarificaton. :)

**Reply by henk23** [reply]
Actually, I think this can be closed as a duplicate of #13944
